### DISCUSSION TITLE
Starting out with Orchard Core CMS. Recipes vs Themes. Migrations

### DISCUSSION THREAD

**Discussion by happygrizzly** [OrchardCMS/OrchardCore#9579] [Q&A] [answered]
Hi everyone! 

- What's the (essential) difference between recipes and themes? Are there any limitations / conventions applied to them?
- recipes.json allows us to bootstrap the app with content types, workflows etc (serves as the first documented change). When and why do we need migrations? How doesn't it contradict with user being able to change content via UI?

**Comment by Skrypt** [comment, marked as answer]
I recipe can be bundled as part of a Theme because a Theme is a module.
This way, when you run a tenant setup you can choose to apply some basic content creation that are necessary for your Theme templates to work.

A migration is needed when you require to either create, alter, update data of a database table, index. A migration can be necessary for example when you decide to change the structure of a Content Type and want to migrate the existing data using the new structure.

Not sure about the contradiction here since recipes should only be available for admins to be executed. It's not a common user feature.

**Reply by happygrizzly** [reply]
I haven't thought about the recipes execution. So there's no contradiction) 
Thank you!
